agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -713,3 +713,40 @@ func TestNetworkModeBridge(t *testing.T) {
 		t.Fatalf("Networking mode bridge testing failed, err: %v", err)
 	}
 }
+
+// TestDockerConcurrentPull tests the agent can perform concurrent pull for docker >= 1.9.0
+// Start 4 tasks totally 40 containers at the same time.
+func TestDockerConcurrentPull(t *testing.T) {
```

Comment:
In general, I think this test is a really good candidate for improving the `simpletests` generator.  It doesn't currently support multi-task tests, but that shouldn't be terribly hard to add.  I don't think it's necessary to do so for this PR, but if you don't want to handle it now we should make an issue to track it.